test/mako/alerter/slack/message.go

```diff
@@ -52,30 +61,65 @@ func Setup(userName, tokenPath, repo string, dryrun bool) (*messageHandler, erro
 	if config == nil {
 		return nil, fmt.Errorf("no channel configuration found for repo %v", repo)
 	}
-	return &messageHandler{client: client, config: *config, dryrun: dryrun}, nil
+	return &MessageHandler{
+		readClient:  readClient,
+		writeClient: writeClient,
+		config:      *config,
+		dryrun:      dryrun,
+	}, nil
 }
 
-// Post will post the given text to the slack channel(s)
-func (smh *messageHandler) Post(text string) error {
-	// TODO(Fredy-Z): add deduplication logic, maybe do not send more than one alert within 24 hours?
-	errs := make([]error, 0)
+// SendAlert will send the alert text to the slack channel(s)
+func (smh *MessageHandler) SendAlert(text string) error {
+	dryrun := smh.dryrun
 	channels := smh.config.channels
-	mux := &sync.Mutex{}
+	errCh := make(chan error)
+	defer close(errCh)
 	var wg sync.WaitGroup
```

Comment:
I know this should work, I think explicitly closing errCh after `wg.Wait` is nicer, so that the goroutine doesn't spin to read the channel that will never be used again

Reply:
The previous code had a bug, this PR fixed it.
Also added a simple unit test.